### ISSUE TITLE
feature request: allow setting the current step without calling `logger.report_*`

### ISSUE THREAD

**Issue by patzm** [allegroai/clearml#440]
Currently, automatic reporting (e.g. CPU / GPU / memory utilization) relies on the last reported step when `logger.report_*` was called. If that doesn't happen for some time, it falls back to using milliseconds since start for the x-axis: https://github.com/allegroai/clearml/blob/5a9155b2039413280f13dfded1121470c4c4323d/clearml/utilities/resource_monitor.py#L110-L111

This is problematic in a few scenarios:
1. the user doesn't want to artifically create dummy reports and the first _desired_ report (if any) would take longer than the time-out
1. the worker has died and gets restarted. This is affected by #439.

In both cases, it would be nice to be able to do something like
```python
clearml_task.set_iteration(current_iteration)
```
If the worker resumes and recovers (externally, maybe from a checkpoint on disk) from iteration `433`, we call
```python
clearml_task.set_iteration(433)
```
Or in the default use-case:
```python
for iteration in range(max_steps):
    clearml_task.set_iteration(iteration)
    # do some work
    loss = train_step()
    if iteration % 1000:  # very infrequent reporting
        clearml_task.report_scalar(loss, "loss", iteration)
```

Links:
* automatic logging is configured through [`Task.init`](https://allegro.ai/clearml/docs/rst/references/clearml_python_ref/task_module/task_task.html#clearml.task.Task.init)

**Comment by patzm** [comment]
Yes, for use case 2 I see that point. However, I don't think it would cater to use case 1. (both are described in https://github.com/allegroai/clearml/issues/440#issue-981470132).
WDYT?
Btw I am also open to contribute.

**Comment by bmartinn** [comment]
Can we combine #439 and this one, it looks like the same feature request/use-case , is that correct?

> Btw I am also open to contribute.

Thank you! ❤️  PRs are much appreciated, and usually are merged relatively quickly :)

**Comment by patzm** [comment]
It is slightly different. If we follow through with your proposal in https://github.com/allegroai/clearml/issues/439#issuecomment-907882223 (either _(1)_ or _(2)_), then I think we would implement two methods:
1. `set_last_checkpoint`
2. `set_iteration`: this is like reporting without a reporting payload.

And maybe also getters for the two.

Regarding contribution: could you point me to where I should start looking and contribution instructions. Ideally I can only work on client-side code. I think setting up the server backend as well would be too much. I don't have much experience with it and also not a _lot_ of time :wink:.

**Comment by bmartinn** [comment]
> Regarding contribution: could you point me to where I should start looking and contribution instructions.

https://github.com/allegroai/clearml/blob/master/docs/contributing.md

> I can only work on client-side code. 

Client SDK will be great! (server-side is definitely more complicated to PR to :)
